index.js

```diff
@@ -215,72 +157,68 @@ function Server (opts) {
 }
 
 Server.prototype._onError = function (err) {
-  var self = this
-  self.emit('error', err)
+  this.emit('error', err)
 }
 
 Server.prototype.listen = function (/* port, onlistening */) {
-  var self = this
-
-  if (self._listenCalled || self.listening) throw new Error('server already listening')
-  self._listenCalled = true
+  if (this._listenCalled || this.listening) throw new Error('server already listening')
+  this._listenCalled = true
 
-  var lastArg = arguments[arguments.length - 1]
-  if (typeof lastArg === 'function') self.once('listening', lastArg)
+  const lastArg = arguments[arguments.length - 1]
+  if (typeof lastArg === 'function') this.once('listening', lastArg)
 
-  var port = toNumber(arguments[0]) || arguments[0] || 0
+  const port = toNumber(arguments[0]) || arguments[0] || 0
 
   debug('listen (port: %o)', port)
 
   function isObject (obj) {
     return typeof obj === 'object' && obj !== null
   }
 
-  var httpPort = isObject(port) ? (port.http || 0) : port
+  const httpPort = isObject(port) ? (port.http || 0) : port
 
-  if (self.http) self.http.listen(httpPort)
+  if (this.http) this.http.listen(httpPort)
 }
 
 Server.prototype.close = function (cb) {
-  var self = this
   if (!cb) cb = noop
   debug('close')
 
-  self.listening = false
-  self.destroyed = true
+  this.listening = false
+  this.destroyed = true
 
-  if (self.ws) {
+  if (this.ws) {
     try {
-      self.ws.close()
+      this.ws.close()
     } catch (err) {}
   }
 
-  if (self.http) self.http.close(cb)
+  if (this.http) this.http.close(cb)
   else cb(null)
 }
 
-Server.prototype.createSwarm = function (infoHash, cb) {
+Server.prototype.createSwarm = function (infoHash) {
   if (Buffer.isBuffer(infoHash)) infoHash = infoHash.toString('hex')
 
-  var swarm = this.torrents[infoHash] = new Swarm(infoHash, this)
-  cb(null, swarm)
+  const swarm = this.torrents[infoHash] = new Swarm(infoHash, this)
+  return swarm
 }
 
 Server.prototype.deleteSwarm = function (infoHash) {
-  var self = this
+  const self = this
   process.nextTick(function () {
     delete self.torrents[infoHash]
   })
 }
 
-Server.prototype.getSwarm = function (infoHash, cb) {
+Server.prototype.getSwarm = function (infoHash) {
   if (Buffer.isBuffer(infoHash)) infoHash = infoHash.toString('hex')
 
-  cb(null, this.torrents[infoHash])
+  return this.torrents[infoHash]
```

Comment:
We made this method asynchronous so that people could replace `getSwarm` with their own implementation that talked to an async backend, like a database or something. So even though it appears pointless in the default in-memory implementation, it is actually useful.